api/app/Models/PoolCandidate.php

```diff
@@ -87,166 +87,109 @@ public function scopeClassifications(Builder $query, ?array $classifications): B
-        $query->whereExists(function ($query) use ($operationalRequirements) {
-            $query->select('id')
-            ->from('users')
-            ->whereColumn('users.id', 'pool_candidates.user_id')
-            ->whereJsonContains('accepted_operational_requirements', $operationalRequirements);
+        // point at filter on User
+        $query->whereHas('user', function ($query) use ($workRegions) {
+            User::scopeLocationPreferences($query, $workRegions);
         });
 
         return $query;
     }
-    public function filterByLocationPreferences(Builder $query, ?array $workRegions): Builder
+    public function scopeLocationPreferences(Builder $query, ?array $workRegions): Builder
     {
         if (empty($workRegions)) {
             return $query;
         }
 
-        // WorkRegion acts as an OR filter. The query should return candidates willing to work in ANY of the regions.
-        $query->whereExists(function ($query) use ($workRegions) {
-            $query->select('id')
-            ->from('users')
-            ->whereColumn('users.id', 'pool_candidates.user_id')
-            ->where(function ($query) use ($workRegions) {
-                foreach ($workRegions as $index => $region) {
-                    if ($index === 0) {
-                        // First iteration must use where instead of orWhere
-                        $query->whereJsonContains('location_preferences', $region);
-                    } else {
-                        $query->orWhereJsonContains('location_preferences', $region);
-                    }
-                }
-            });
+        // point at filter on User
+        $query->whereHas('user', function ($query) use ($workRegions) {
+            User::scopeLocationPreferences($query, $workRegions);
         });
+
         return $query;
     }
-
-    public function filterByLanguageAbility(Builder $query, ?string $languageAbility): Builder
+    public function scopeLanguageAbility(Builder $query, ?string $languageAbility): Builder
     {
         if (empty($languageAbility)) {
             return $query;
         }
 
         // point at filter on User
         $query->whereHas('user', function ($query) use ($languageAbility) {
-            User::filterByLanguageAbility($query, $languageAbility);
+            User::scopeLanguageAbility($query, $languageAbility);
         });
+
         return $query;
     }
-
-    public static function filterByPools(Builder $query, ?array $pools): Builder
+    public static function scopeAvailableInPools(Builder $query, ?array $poolIds): Builder
     {
-        if (empty($pools)) {
+        if (empty($poolIds)) {
             return $query;
         }
 
         // Pool acts as an OR filter. The query should return candidates in ANY of the pools.
-        $poolIds = [];
-        foreach ($pools as $pool) {
-            array_push($poolIds, $pool['id']);
-        }
```

Comment:
If you look at the schema, the data passed to filterByPools is in the format `{id: ID!}[]`, meaning some kind of step like this is required to flatten it into an array of ids instead of an array of objects which hold ids.

Reply:
Hmm, this must of been an accident 😆 I don't remember removing this.

Reply:
Actually, I needed to update this since the ApplicantFilterInput uses the `@pluck(key: "id")` scope. I added it to the PoolCandidateInput scope as well.